syntax = "proto3";

package mafia;

service Mafia {
    rpc Connect(Request) returns (Response) {}
    rpc ChooseRoom(ChooseRoomRequest) returns (Response) {}
    rpc SetName(Request) returns (Response) {}
    rpc WaitStart(Request) returns (stream MemberResponse) {}
}

enum Status {
    SUCCESS = 0;
    FAIL    = 1;
}

message Request {
    string message = 1;
}

message ChooseRoomRequest {
    int32 room = 1;
}

message Response {
    Status status = 1;
    string message = 2;
}

message MemberResponse {
    int32 unnamed = 1;
    string connected = 2;
    Response response = 3;
}
